lib/analyse.ml

```diff
@@ -244,10 +244,14 @@ module Analysis = struct
     let ( // ) = Fpath.( / ) in
     Fpath.v "packages" // name // (name^"."^version) // "opam"
 
-  let add_package_data ~dir (pkg, kind) packages =
-    match packages with
-    | Error _ as err -> Lwt.return err
-    | Ok packages ->
+  let add_package_data ~dir ~job (pkg, kind) packages =
+    match (packages, kind) with
+    | Error _ as err, _ ->
+      Lwt.return err
+    | Ok packages, Deleted ->
+      Current.Job.log job "Skipping deleted package: %s" (OpamPackage.to_string pkg);
+      Lwt.return_ok packages
```

Comment:
I think this fix makes sense at the analysis stage. Crashing with an internal error on this case is not good behavior!